e2e/test/Helpers/logging/TestLogger.cs

```diff
@@ -27,28 +25,22 @@ public class TestLogger
 
         public IDictionary<string, string> Properties { get; } = new Dictionary<string, string>();
 
-        protected TestLogger()
+        // Static constructor is called at most one time, before any instance constructor is invoked or member is accessed.
+        static TestLogger()
```

Comment:
I preferred using a static constructor over adding locks to an instance constructor, since the implementation requires this block to be executed only once, and this was an easy way to achieve this.